forwards_paper.tex

```diff
@@ -822,46 +822,42 @@ \subsubsection*{Computational complexity}
-$2NT$ nodes and
-$4NT$ edges.
-Suppose that $T$ is large enough that all samples coalesce within the simulation with high probability,
-($T = 20N$, say).
+$NT$ nodes and
+$2NT$ edges.
+Suppose that $T$ is large enough that all samples coalesce within the simulation with high probability
+($T = 20N$, say, giving a total space requirement of $60N^2$).
 After simplification, we are left with the tree sequence describing the history
-of only the current generation of $2N$ individuals.
-If no coalescent events involve more than two lineages, this tree sequence has $4N-2$ edges to describe the leftmost tree;
+of only the current generation of $N$ individuals.
+If no coalescent events involve more than two lineages, this tree sequence has $2N-2$ edges to describe the leftmost tree;
 and each time the marginal tree changes along the sequence,
 four edges end and four new edges begin
 (except for changes affecting the root, which require fewer; see \citet{kelleher2016efficient}).
 Coalescent theory tells us that
-the expected total length of the edges of a marginal tree is approximately $4N\log(2N)$.
+the expected total length of the edges of a marginal tree is approximately $2N\log(N)$.
 Since we have taken one crossover per individual per generation (as is typical for a single chromosome),
 and we are measuring tree lengths in units of generations,
-the mean number of ancestral recombination events that occur somewhere on the marginal trees
-per unit of genetic length.
+this is the mean number of ancestral recombination events that occur somewhere on the marginal trees.
 Any such ancestral recombination might change the marginal gene tree;
 if it does, it will add four new edges.
 Not all such recombinations actually change the tree topology,
-so at most we expect $16N\log(2N)$ new edges after those required to describe the first tree.
+so at most we expect $8N\log(N)$ new edges after those required to describe the first tree.
 Similarly, not every new edge derives from a never-before-seen node,
 but the number of nodes is at most equal to the number of edges plus the sample size.
-This reduces the initial storage of $120 N^2$ items to
-% 4N - 2 + 2N + 4N log(2N) * 4 * 2 = 6N + 32 N log(2N)
-the space required after simplification of roughly
-\begin{align}
-    6N + 32 N \log(2N) .
-\end{align}
-At $N=10^4$, this reduces the space required to store the history of the entire population
-by a factor of 3,700.
+The space complexity after simplification is therefore $O(N + N\log N)$, since
+we need $O(N)$ space to store the first marginal tree, and $O(1)$ space
+for each of the $O(N \log N)$ trees that follow. Thus, we need $O(N^2)$ space
+to store the complete history of the simulation, but only $O(N + N \log N)$
+space to store the history that is relevant to the extant population.
 
 %% Progressive memory reduction as a function of T
 \citet{padhukasahasram2008exploring} obtained impressive speedups relative to contemporary simulators
```

Comment:
I haven't gone any farther with changing 2N to N here. I've spent a good while going through the calculation and can't follow the argument, so I left it alone.

Reply:
I will add more explanation.

One thing we should do is add a citation: the basic calculation here is a rough version of calculating the total length of branches in a coalescent tree up until time T, which gives the distribution of the age of a derived allele.  I've tried to find the right paper to cite, and not found it; but we could cite Wakeley's book for more background.

Reply:
I think that Watterson 1975 may be that citation.  There are several "stealth" coalescent arguments there, which lead up to the correction for the expected number of mutations in the entire pop vs in a sample.  Also, Kingman's paper on the coalescent justifies the usual first-order approximation by showing that the process for a sample converges in distribution to that of the population.  (I think I'm remembering Kingman correctly...)

One issue w/citing Wakeley's book is that it is no longer in print, which is too bad, as it is quite readable.

Reply:
Gee, it does do that sort of calculation, although not exactly.  That's a nice paper; I don't think I'd read it before!

Reply:
I'm fine with citing Wakeley's book. It should be in print, it's great!